tests/integration/ci-runner.py

```diff
@@ -626,7 +626,7 @@ def run_test_group(
             info_basename = test_group_str + "_" + str(i) + ".nfo"
             info_path = os.path.join(repo_path, "tests/integration", info_basename)
 
-            test_cmd = " ".join([test for test in sorted(test_names)])
```

Comment:
Have you tested this? I guess this will not help, since they will be passed to the runner command as `"foo` `bar"`, see #51514 for a fix that should work

Reply:
Your fix is better, and we will apply it.
But I don't understand your point - why it will be passed in a wrong way?

Also, this PR contains multiple other fixes - the code is written in the wrong culture.

Reply:
For the same reason as this does not work correctly:

```
$ echo 'echo \"foo     bar\"' | xargs -i -t sh -c "{}"
sh -c 'echo "foo     bar"'
foo     bar
```

Here is a comment for one of patches in my PR:

>Note, that I've replaced PYTEST_OPTS with handled by pytest itself
>PYTEST_ADDOPTS, since "sh -c $PYTEST_OPTS" does not allows you to
>preserve the arguments correctly.